c10/CMakeLists.txt

```diff
@@ -41,6 +41,28 @@ if (${COMPILER_SUPPORTS_HIDDEN_VISIBILITY})
   target_compile_options(c10 PRIVATE "-fvisibility=hidden")
 endif()
 
+# ---[ Check for NUMA support
+if (USE_NUMA)
```

Comment:
@ilia-cher can we just move this check from caffe2 to c10 like this? and do we need to do anything for `USE_NUMA`?

Reply:
cc. @orionr @pietern  to check CMake changes

Reply:
cc @Yangqing

Reply:
I would remove C10_IS_NUMA_AVAILABLE and C10_DISABLE_NUMA and simply update the USE_NUMA in cmake/Dependencies.cmake to do these CHECK_CXX_SOURCE_COMPILES  include checks if needed.